pkg/providers/azure.go

```diff
@@ -319,21 +328,52 @@ func (sc *SetUpCmd) createFederatedCredentials() error {
 
 }
 
-func (sc *SetUpCmd) getAppObjectId() error {
+func (wrapper *AzureCredentialWrapper) OAuthToken() string {
+	decoded, err := base64.StdEncoding.DecodeString(wrapper.token)
+	if err != nil {
+		fmt.Println("decode error:", err)
```

Comment:
if there's an error or possibility of error we should have `error` be in the function signature and pass it around appropriately. 

I see that we pass this struct into a function we don't own so you might have to push the error upwards. Create a constructor for the struct and catch the error in the constructor.